app/uk/gov/hmrc/gform/form/FormService.scala

```diff
@@ -47,11 +51,11 @@ class FormService[F[_]: Monad](formPersistence: FormPersistenceAlgebra[F], fileU
     val formId = FormId(userId, formTemplateId, accessCode)
     val expiryDate = timeProvider.localDateTime().plusDays(expiryDays)
 
-    for {
-      envelopeId <- fileUpload.createEnvelope(formTemplateId, expiryDate)
-      form = Form(
+    if (config.env == "OFSTED") {
```

Comment:
This `if ... else` should be completely removed and creation of `envelopeId` should be put as a dependency instead. In this way `FormService` may stay oblivious to what environment it runs in.